pandas/core/internals.py

```diff
@@ -5598,8 +5598,10 @@ def get_reindexed_values(self, empty_dtype, upcasted_na):
                     if len(values) and values[0] is None:
                         fill_value = None
 
-                if getattr(self.block, 'is_datetimetz', False):
-                    pass
+                if getattr(self.block, 'is_datetimetz', False) or \
```

Comment:
so I would like to fix this more generally.

define something like this on the Block. This *should* just work generally
```
def empty(self):
    """ return a same shape block filled with the empty value for this block"""
   arr = np.full(np.prod(self.shape), self._na_value)
   arr = _block_shape(arr, self.ndim)
   return self.make_block_same_klass(arr)
```

if can get this to work , then can clean up a bunch of additional code.

Reply:
Not sure how this simplifies. Won't you have to cast the result to the empty_dtype anyway, which doesn't necessarily match the dtype of the block?

Reply:
What would this do for non-NA holding blocks? Raise a TypeError probably?

Reply:
And should we pass through `placement`?

Reply:
Actually, perhaps something like

```
def empty(self, dtype=None):
    ...
```

Then you get back a block for `dtype`. By defualt you get the same type.

This solves two problems:

1. you can do `IntBlock.empty(dtype=float)` just fine. (will still raise if that type can't hold NA)
2. As @paul-mannino indicated, we have to cast to empty_dtype anyway.